### PR TITLE
Update the example to work with the v0 endpoints

### PR DESCRIPTION
**Pull request by ChadKillingsworth**
Adds support for the v0 OIDC flow. Requires redirecting the user back through the OIDC provider. Includes support for PKCE.

### CODE REVIEW

**Comment by DevWithTheHair** [comment]
@samuelcouch let's independently confirm that this works as we expect, and assuming it does we can merge as-is.

We can use this PR to note the items that we would like to change ((e.g. clarifying comments, code structure to match our style, that sort of stuff) so that we have an easy reference list of things that we can update in a fast-follow PR.

**Review by samuelcouch — APPROVED** [review]
Tested, looks good to me 👍

**Comment by DevWithTheHair** [comment]
Seems good to merge as I've independently tested this with a new plugin setup and it looks like it is working for me.

This is effectively a follow-up to https://github.com/Banno/simple-plugin-example/pull/2 (for records purposes).

**Review by DevWithTheHair — APPROVED** [review]
Looks good to merge, I'll add some feedback items that Sam and I can address in a future PR.

File: server.js [hunk 1]
```diff
@@ -29,20 +30,47 @@ app.get('/static', (req, res) => {
     res.render('pages/static')
 })
 
+const stateStore = new Map()
```
Comment (DevWithTheHair) [inline server.js:33]: **future:** We'll want to add a comment here with caveats about how this isn't something you'll want to do in Production (i.e. it's only done here in favor of _simplicity_ of the example code).

I'm thinking something similar to what we did for the other example project a la https://github.com/Banno/consumer-api-openid-connect-example/blob/91f7e0febbc5d87fe33d37345c2d5520480138a2/server.js#L39-L42

File: server.js [hunk 2]
```diff
@@ -29,20 +30,47 @@ app.get('/static', (req, res) => {
...
+            .replace(/=/g, '')
+            .replace(/\+/g, '-')
+            .replace(/\//g, '_')
+
+        state = crypto.randomBytes(60).toString('hex').slice(0, 128)
+        stateStore.set(state, codeVerifier)
+        res.cookie(`STATE_${state}`, codeVerifier, {httpOnly: true, sameSite: 'lax'})
+
+        // redirect
+        res.redirect(`${config.api.environment}/a/consumer/api/v0/oidc/auth?scope=openid&response_type=code&client_id=${config.api.client_id}&redirect_uri=${encodeURIComponent(REDIRECT_URI)}&state=${state}&code_challenge=${CODE_CHALLENGE}&code_challenge_method=S256`)
```
Comment (DevWithTheHair) [inline server.js:61]: **future:** This whole chunk probably needs some clarifying comments as it is a bit non-obvious what this is doing here unless the reader is very knowledgeable about OAuth+OpenID Connect in general and PKCE in particular.

We might also want a link to something like the Authentication (Command Line) Quickstart, although that's just a quick thought. https://jackhenry.dev/open-api-docs/consumer-api/quickstarts/AuthenticationCommandLine/

**Pull request merged**